ci/jenkins/jobs/macros.yaml

```diff
@@ -102,6 +102,7 @@
       - shell: |-
           #!/bin/bash
           set -ex
+          ./ci/jenkins/docker_login.sh --docker-user "${{DOCKER_USERNAME}}" --docker-password "${{DOCKER_PASSWORD}}"
           DOCKER_REGISTRY="$(head -n1 ci/docker-registry)"
```

Comment:
it's a little bit strange that we call `docker_login.sh` unconditionally, and then we run `DOCKER_REGISTRY`. In theory, if we have `DOCKER_REGISTRY="$(head -n1 ci/docker-registry)"`, that means that `DOCKER_REGISTRY` is not guaranteed to be `docker.io`, but `docker_login.sh` assumes that the registry is `docker.io`.

Maybe these 2 instructions should be swapped, and `docker_login.sh` should return immediately if `DOCKER_REGISTRY` is not `docker.io`? Or some other solution that makes `docker login` conditional.

Reply:
Yes it make sense, I add a registry flag for the login script.